### DISCUSSION TITLE
AuthenticationState(ClaimPrincipal) - exceeds message limit size when more than 12 claims

### DISCUSSION THREAD

**Discussion by azraeana** [dotnet/aspnetcore#54881] [Q&A]
Hi!

We have to utilize a CustomAuthenticationStateProvider for our Blazor Server App  due to pre-existing security tables and structure. In the override of the GetAuthenticationState of my provider, I build a claims principal that has an identity with several claims.  We cannot do role based authorization because our organization allows two users of the same role to have customized permissions. 

So when attempting to return a claims identity with 13 claims (in the format of key: permission name, value: true) signalr breaks and says that the message size of 32kb was exceeded.  

Is there a better way to handle permission driven authorization?  Or is my only option increasing the message size limit?  We use authorize views in our UI and would love to continue utilizing that instead of a bunch of @ if blocks.

**Comment by engineering87** [comment]
Hi there,
consider slightly increasing the SignalR message size limit:

```csharp
services.AddSignalR(hubOptions =>
{
    hubOptions.MaximumReceiveMessageSize = 64 * 1024; // Increase to 64KB
});
```
